app/src/main/java/org/permanent/permanent/ui/archiveOnboarding/compose/ArchiveOnboardingScreen.kt

```diff
@@ -130,10 +140,79 @@ fun ArchiveOnboardingScreen(
         )
     }
 
+    val priorities = rememberSaveable(
+        saver = listSaver(
+            save = {
+                it.map { priority ->
+                    listOf(
+                        priority.type.ordinal,
+                        priority.description,
+                        priority.isChecked.value
+                    )
+                }
+            },
+            restore = { restoredPriorities ->
+                restoredPriorities.map {
+                    OnboardingPriority(
+                        type = OnboardingPriorityType.values()[it[0] as Int],
+                        description = it[1] as String,
+                        isChecked = mutableStateOf(it[2] as Boolean)
+                    )
+                }.toMutableStateList()
+            }
+        )
+    ) {
+        mutableStateListOf(
```

Comment:
This screen became kinda large. Maybe it makes sense to move the priorities and goals creation either to a viewmodel or to a builder class?